src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -949,23 +945,19 @@ private void joinTokenRing(int delay, boolean autoBootstrap, Collection<String>
             }
             setMode(Mode.JOINING, "waiting for ring information", true);
             // first sleep the delay to make sure we see all our peers
-            for (int i = 0; i < delay; i += 1000)
-            {
-                // if we see schema, we can proceed to the next check directly
-                if (!Schema.instance.getVersion().equals(Schema.emptyVersion))
-                {
-                    logger.debug("got schema: {}", Schema.instance.getVersion());
-                    break;
-                }
-                Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
-            }
+            Uninterruptibles.sleepUninterruptibly(delay, TimeUnit.MILLISECONDS);
```

Comment:
I think having some delay here is still important because we know from experience that you should wait at least some amount of time here before proceeding. The intention of this change is to make the bootstrapping node wait 
- at least 30s
- wait longer if it thinks the schema is consistent

A bad scenario would be if we get the non-empty schema by gossiping with a node in a network partition, think we have gossiped with everyone because we haven't waited long enough, and then pass the subsequent checks.

Reply:
does that risk exist even if we wait for `isSchemaConsistent`? in that case, we should make our check more strict instead of just relying on a 30 second wait to hope it's safe